packages/shell-api/src/shell-internal-state.ts

```diff
@@ -167,7 +167,7 @@ export default class ShellInternalState {
 
     const setFunc = (newDb: any): Database => {
       if (getShellApiType(newDb) !== 'Database') {
-        throw new MongoshInvalidInputError('Cannot reassign \'db\' to non-Database type');
+        throw new MongoshInvalidInputError('Cannot reassign \'db\' to non-Database type', CommonErrors.InvalidArgument);
```

Comment:
Possible nit - it's a bit odd to have an error `InvalidArgument` for something that's not an argument. However I'm not sure what else to call, so we can keep this as Argument if you think its' the closest fit. It's inconsequential since we don't really need telemetry on this specific error

Reply:
Ah I see - since it's an assignment. I switched it to `InvalidOperation`.